server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -396,8 +367,32 @@ void executeRequest(
                     searchPhaseProvider.apply(delegate)
                 );
             } else {
+                if ((listener instanceof TelemetryListener tl) && CCS_TELEMETRY_FEATURE_FLAG.isEnabled()) {
```

Comment:
are these additional instanceof checks necessary? Could we instead have the relevant listeners implement the new interface and be no-op, or something along those lines?

Reply:
Yes they are necessary because `executeRequest` can also be called from `shouldOpenPIT` branch (line 496) where it's a different listener. I mean I could make that listener implement TelemetryListener too, but that would not be a good idea since on that code path, we should not collect the telemetry as I understand, so we still will need the `if` conditional and somehow know in which code path are we, if I understand it correctly.

Reply:
Or maybe I am wrong and we should collect the telemetry there? I am a bit confused on the whole PIT topic right now, so I am not 100% sure what is supposed to happen in that code branch...

Reply:
There's also a call to executeRequest from `TransportOpenPointInTimeAction` so I think we'd still need the check here one way or another.